zulip_bots/zulip_bots/bots/trivia_quiz/trivia_quiz.py

```diff
@@ -198,16 +200,41 @@ def format_quiz_for_markdown(quiz_id, quiz):
     )
     return content
 
-def grade_question(quiz, answer):
-    correct = (answer == quiz['correct_letter'])
+def update_quiz(quiz: Dict[str, Any], quiz_id: str, bot_handler: Any) -> None:
+    bot_handler.storage.put(quiz_id, json.dumps(quiz))
 
-    if correct:
-        long_answer = quiz['answers'][answer]
-        response = '**CORRECT!** {long_answer} :tada:'.format(long_answer=long_answer)
-        return correct, response
+def handle_answer(quiz: Dict[str, Any], answer: str, quiz_id: str,
+                  bot_handler: Any) -> Tuple[bool, str]:
+    long_answer = quiz['answers'][quiz['correct_letter']]
+    start_new_question = False
+
+    if answer not in quiz['answered_options'] and not quiz['ended']:
+        # The question has not been ended, so we store all the options
+        # which are being answered
+        quiz['answered_options'].append(answer)
+        update_quiz(quiz, quiz_id, bot_handler)
+
+    if answer == quiz['correct_letter']:
+        response = '**CORRECT!** {long_answer} :tada:'
+        # We don't start a new question if the question has already been
+        # answered
+        start_new_question = not quiz['ended']
+    else:
+        if len(quiz['answered_options']) == 3:
+            # We reveal the answer and start a new question upon having three failed
+            # attempts at answering a question
+            response = '''Three failed attempts at {quiz_id}:disappointed:
```

Comment:
MINOR: This is personal style (I prefer to avoid triple-quotes, at least when wrapping onto the next line), so does not require changing, but you could consider exploring other styles such as:
* multiple single-quoted strings, grouped with parentheses into one value and enabling splitting over lines;
* If the `"\n"` become too much noise, then I favour using `.join` on a list of strings, each to be a separate line.

Reply:
I followed the triple quotes this has been followed throughout the script and I wanted to be consistent. Here is the [example](https://github.com/zulip/python-zulip-api/blob/master/zulip_bots/zulip_bots/bots/trivia_quiz/trivia_quiz.py#L188-L196).